src/twisted/test/test_defgen.py

```diff
@@ -338,6 +339,58 @@ def _noYield():
 
         self.assertIn("inlineCallbacks", str(self.assertRaises(TypeError, _noYield)))
 
+    def test_internalDefGenReturnDoesntLeak(self):
+        """
+        When one inlineCallbacks calls another, the internal L{_DefGen_Return}
+        flow control exception doesn't leak into tracebacks captured in the
+        caller.
+        """
+        clock = task.Clock()
+
+        @inlineCallbacks
+        def _returns():
+            yield task.deferLater(clock, 0)
+            returnValue(6)
+
+        @inlineCallbacks
+        def _raises():
+            try:
+                yield _returns()
+                raise TerminalException("boom")
+            except TerminalException:
+                return traceback.format_exc()
+
+        d = _raises()
+        clock.advance(0)
+        tb = self.successResultOf(d)
+        self.assertNotIn("_DefGen_Return", tb)
```

Comment:
can we also add a "positive" assertion about how to traceback should look like ?

If the traceback is empty, the test will still pass...but maybe we have other tests that will fail...so no need for extra assertions here.

```suggestion
        # The internal function is in the traceback.
        self.assertNotIn("_DefGen_Return", tb)
        # No other extra exception is in the traceback.
        self.assertNotIn(
            "During handling of the above exception, "
            "another exception occurred",
            tb)
        # Our targeted exception is in the traceback
        self.assertIn('test_defgen.TerminalException: boom returnValue', tb)
```